airbyte-integrations/connectors/source-stripe/source_stripe/streams.py

```diff
@@ -159,12 +159,84 @@ def get_start_timestamp(self, stream_state) -> int:
 
         return start_point
 
+class IncrementalStripeStreamWithUpdates(IncrementalStripeStream):
+    """
+        This is a base class for incremental streams that support updates using the events API
+        In first sync it will not get any updates as the records are already updated
+        After first sync it will get all updates starting from date of the last sync
+    """
+    event_types = None
+    update_field = "event_created"
+
+    def read_records(self,stream_slice, stream_state, **kwargs) -> Iterable[Mapping[str, Any]]:
+        # If this is not the first sync also get the updates
+        if stream_state.get(self.cursor_field) is not None: 
+            newRecordsIDList =[]
+            for record in super().read_records(stream_slice=stream_slice, stream_state=stream_state, **kwargs):
+                newRecordsIDList.append(record[self.get_record_id_key(record)])
```

Comment:
Please check this comment https://www.notion.so/Stripe-updated-records-d1df062071b848cb87966c1b2d56f8a1?d=b7322691fdc243c0a6a8e070261d1830#c4f80f7cf03c4050b382cddbf99a8786 about `get_record_id_key`. As in with `event_created` possibly being same for multiple events records (some time no change in record and sometime some changes), we might be better keep this append only resource in wh table to avoid any data discrepancy.

This could have been ok, thing is sometimes final value of the record might get overriden with stale data causing data error. So keep it append only and let customer define how they want to find lates value should be better. We did share this to some customer and they were able to put some hack to be able to identify latest record basd on some checks on values of some fields.